packages/pigeon/lib/swift_generator.dart

```diff
@@ -607,6 +589,75 @@ import FlutterMacOS
     indent.newln();
   }
 
+  /// Writes decode and casting code for any type.
+  ///
+  /// Optional parameters are necessary for class decoding only.
```

Comment:
It's more subtle than that; I'm pretty sure that passing `customClassNames` would actively break things (creating essentially the bug I fixed for C++ in https://github.com/flutter/packages/pull/3573). I was actually confused at first about how the PR *wasn't* causing a regression there until I noticed that the call sites weren't passing the class list.

If I'm correct about that, I think it would be better to make things much more explicit. E.g., you could rename `customClassNames` to `listEncodedClassNames`, and change this comment to explicitly say that a value for that must be provided only in the case of class decoding, with a link to https://github.com/flutter/flutter/issues/119351 for context. (Alternately you could leave the name, but add an explicit bool like I did in the C++ generator.)

Reply:
I renamed them. And made the comment more specific.